### PR TITLE
integration: Wait for PV to be available in TestVolumeProvision

### LINKED ISSUE #81368: TestVolumeProvision is flaky
**Which jobs are failing**:

pull-kubernetes-integration

**Which test(s) are failing**:

TestVolumeProvision

Example: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/81330/pull-kubernetes-integration/1161315691412852736/

```
--- FAIL: TestVolumeProvision (12.16s)
    volume_binding_test.go:1149: Provisoning annotaion on PVC volume-schedulingfc108048-dc1f-4999-8323-f3302c9eb3ca/pvc-w-canbind not bahaviors as expected: PVC volume-schedulingfc108048-dc1f-4999-8323-f3302c9eb3ca/pvc-w-canbind not expected to be provisioned, but found selected-node annotation
    volume_binding_test.go:1191: PV pv-w-canbind phase not Bound, got Available
```

**Since when has it been failing**:

Not sure, [triage board](https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&test=TestVolumeProvision) and [test grid](https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-integration&sort-by-flakiness=) aren't displaying any results

/sig storage

### PR DESCRIPTION
**Pull request by cofyc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
/kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #81368 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by cofyc** [comment]
/test pull-kubernetes-integration 
(run more times to check if it's stable)

**Comment by cofyc** [comment]
/test pull-kubernetes-integration
(run more times to check if it's stable)

File: test/integration/volumescheduling/volume_binding_test.go [hunk 1]
```diff
@@ -783,35 +783,43 @@ func testVolumeProvision(t *testing.T, storageCapacity bool) {
...
 		}
-		// Create PVs
+
+		t.Log("Creating PVs")
 		for _, pvConfig := range test.pvs {
 			pv := makePV(pvConfig.name, classes[pvConfig.scName].Name, pvConfig.preboundPVC, config.ns, pvConfig.node)
 			if _, err := config.client.CoreV1().PersistentVolumes().Create(context.TODO(), pv, metav1.CreateOptions{}); err != nil {
 				t.Fatalf("Failed to create PersistentVolume %q: %v", pv.Name, err)
 			}
+			if err := waitForPVPhase(config.client, pvConfig.name, v1.VolumeAvailable); err != nil {
```
Comment (msau42) [inline test/integration/volumescheduling/volume_binding_test.go:793]: Does this mean the race exists in a real cluster too?

Comment (cofyc) [reply test/integration/volumescheduling/volume_binding_test.go:793]: It's more like the uncertainty in testing. Our volume binder will try to dynamically provision if no static matching PV is found. But in some scenarios, we expect the volume binder to use the static PV, so we need to wait for the PV to be available. 

I think it's fine in real clusters. In theory, it's possible that some newly created PVs may not be synced to the scheduler or not be in the available state when a PVC is created to request a PV from the storage class. But, normally both the static provisioning and dynamic provisioning are acceptable for users if a storage class supports both of them.

**Comment by cofyc** [comment]
@ahg-g can you add this fix into 1.19 milestone?

**Comment by cofyc** [comment]
/retest

found another issue (https://github.com/kubernetes/kubernetes/issues/93830), which I'm going to fix in https://github.com/kubernetes/kubernetes/pull/93831

**Comment by liggitt** [comment]
/hold

Will fixing https://github.com/kubernetes/kubernetes/issues/93830 remove the flake in this test? To @msau42's question, does this failure represent a real bug? If so, we probably want to leave the test that exposed the issue as-is

**Comment by cofyc** [comment]
> /hold
> 
> Will fixing #93830 remove the flake in this test? To @msau42's question, does this failure represent a real bug? If so, we probably want to leave the test that exposed the issue as-is

yes, #93830 is a bug that was introduced in #93511. It's a real bug which we should [fix in the scheduler code](https://github.com/kubernetes/kubernetes/pull/93831).

This PR fixes an issue in the test that [causes failures](https://github.com/kubernetes/kubernetes/issues/81368#issuecomment-666043508) with or without #93551. 
It's not a real bug. When a storage class supports both static and dynamic provisioning, a PVC can be bound to a static PV or dynamic provisioned. However, in some test cases, we want the PVC to be statically bound, so we must wait for pre-provisioned static PVs to be available first. Fixing this issue will not hide the #93830 issue.

**Comment by liggitt** [comment]
ack, thanks

/hold cancel

**Pull request merged**